global/scripts/parallel.shlib

```diff
@@ -172,3 +185,71 @@ function par_addjob_stdin()
     par_jobinputarray+=("$jobinput")
 }
 
+#Add job to the current stage (FSLSUB, BUILTIN or NONE mode)
+#Inputs:
+#1. parallel modein
+#2. queue name for FSLSUB (local submit if empty)
+#3. command with arguments
+function par_add_job_to_stage()
+{
+    local PARALLEL_MODE=$1
+    shift
+
+    #FSLSUB queue name for FSLSUB mode, should be 'NONE' for other modes.
+    local fslsub_queue=$1
+    shift
+
+    local job submit_cmd cmd=("$@")
+
+    if [ "$PARALLEL_MODE" == "FSLSUB" ]; then #current job is submitted, and added to fsl_job_list        
+        submit_cmd=(fsl_sub)
+        if [[ -n "$fslsub_queue" ]]; then 
+            submit_cmd+=(-q "$fslsub_queue")
+        fi
+        if [[ -n "$par_fslsub_job_waitlist" ]]; then
+            submit_cmd+=(-j "$par_fslsub_job_waitlist")
+        fi
+        submit_cmd+=("${cmd[@]}")        
+        job=$("${submit_cmd[@]}")
+        par_fslsub_job_list+=("$job")
+    elif [ "$PARALLEL_MODE" == "BUILTIN" ]; then          #current job is added to the list of local parallel jobs        
+        par_addjob "${cmd[@]}"
+    else #NONE                                          #current job is executed.
```

Comment:
This should probably sanity check the mode, in case the outer script doesn't.  Having a typo of `BUITIN` run in `NONE` mode due to missing sanity checks isn't what we want.

Reply:
Ok

Reply:
You checked this in `par_finalize_stage`, which is used after this function, but you haven't checked it here.

Reply:
Ok